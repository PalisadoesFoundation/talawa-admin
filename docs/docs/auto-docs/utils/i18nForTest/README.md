[**talawa-admin**](../../README.md)

***

[talawa-admin](../../README.md) / utils/i18nForTest

# utils/i18nForTest
