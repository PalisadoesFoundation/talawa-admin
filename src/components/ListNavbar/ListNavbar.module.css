.navbarbgOrglist {
  height: 60px;
  background-color: white;
  display: flex;
  margin-bottom: 30px;
  z-index: 1;
  position: relative;
  flex-direction: row;
  justify-content: space-between !important;
  box-shadow: 0px 0px 8px 2px #c8c8c8;
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 1030;
}

.titlemodal .logo {
  color: #707070;
  margin-left: 0;
  display: flex;
  align-items: center;
  text-decoration: none !important;
}

.logo img {
  margin-top: 0px;
  margin-left: 10px;
  height: 64px;
  width: 70px;
}

.logo > strong {
  line-height: 1.5rem;
  margin-left: -5px;
  font-family: sans-serif;
  font-size: 19px;
  color: #707070;
}

.navitems > .navlinks {
  cursor: pointer;
  margin-left: 28px;
  line-height: 1rem;
  font-size: 16px;
  color: #707070;
  z-index: 1;
  flex-direction: row;
  display: flex;
}

.navitem {
  text-decoration: none;
  color: #707070 !important;
  transition: smooth;
  padding-bottom: 10px;
}
.navlinks:hover {
  text-decoration: none;
  border-bottom: 3px solid #31bb6b;
  font-weight: 600;
}
.languageBtn {
  font-size: 20px;
  color: #31bb6b;
}
.languageBtn:hover {
  color: #31bb6b;
}
.languageBtn:focus {
  color: #31bb6b;
}
.logoutbtn {
  border: 1px solid #e8e5e5;
  box-shadow: 0 2px 2px #e8e5e5;
  border-radius: 5px;
  background-color: #31bb6b;
  height: 40px;
  font-size: 16px;
  color: white;
  outline: none;
  font-weight: 600;
  cursor: pointer;
  transition: transform 0.2s, box-shadow 0.2s;
  padding: 0 20px;
  margin-right: 10px;
}

@media only screen and (max-width: 600px) {
  body::before {
    height: 118px;
  }
  .navbarbgOrglist {
    height: auto;
  }
  .navitems > .navlinks {
    margin-top: 11px;
    margin-left: 0;
  }
  .logoutbtn {
    margin-top: 10px;
  }
}
