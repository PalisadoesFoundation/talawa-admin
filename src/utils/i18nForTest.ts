import i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import HttpApi from 'i18next-http-backend';

import { languageArray } from './languages';
import translationEnglish from '../../public/locales/en.json';

i18n
  .use(LanguageDetector)
  .use(HttpApi)
  .use(initReactI18next)
  .init({
    fallbackLng: 'en',
    supportedLngs: languageArray,
    detection: {
      order: ['cookie', 'htmlTag', 'localStorage', 'path', 'subdomain'],
      caches: ['cookie'],
    },
    resources: {
      en: {
        translation: translationEnglish,
      },
    },
    react: { useSuspense: false },
  });

export default i18n;
