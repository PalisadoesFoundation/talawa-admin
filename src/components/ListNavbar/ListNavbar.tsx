import React from 'react';
import { Nav, Navbar } from 'react-bootstrap';
import { Link, NavLink } from 'react-router-dom';
import i18next from 'i18next';
import Cookies from 'js-cookie';
import { useTranslation } from 'react-i18next';

import styles from './ListNavbar.module.css';
import Logo from 'assets/talawa-logo-200x200.png';
import { languages } from 'utils/languages';

const ListNavbar = () => {
  const { t } = useTranslation('translation', { keyPrefix: 'listNavbar' });

  const isSuperAdmin = localStorage.getItem('UserType') !== 'SUPERADMIN';

  const currentLanguageCode = Cookies.get('i18next') || 'en';

  const logout = () => {
    localStorage.clear();
    window.location.replace('/');
  };

  return (
    <Navbar className={styles.navbarbgOrglist} expand="xl">
      <Navbar.Brand className={styles.navbarBrand}>
        <Link className={styles.logo} to="/orglist">
          <img src={Logo} />
          <strong>{t('talawa_portal')}</strong>
        </Link>
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="navbarScroll" />
      <Navbar.Collapse id="navbarScroll">
        <Nav className="me-auto">
          {!isSuperAdmin && (
            <>
              <Nav.Item className={styles.navitems}>
                <Nav.Link
                  as={NavLink}
                  to="/roles"
                  className={styles.navlinks}
                  activeClassName={styles.navlinks_active}
                >
                  {t('roles')}
                </Nav.Link>
              </Nav.Item>
              <Nav.Item className={styles.navitems}>
                <Nav.Link
                  as={NavLink}
                  to="/requests"
                  className={styles.navlinks}
                  activeClassName={styles.navlinks_active}
                >
                  {t('requests')}
                </Nav.Link>
              </Nav.Item>
            </>
          )}
        </Nav>
        <Nav className="ml-auto">
          <div className="dropdown mr-4">
            <button
              className={`btn dropdown-toggle ${styles.languageBtn}`}
              type="button"
              data-toggle="dropdown"
              aria-expanded="false"
              title="Change Langauge"
            >
              <i className="fas fa-globe"></i>
            </button>
            <div className="dropdown-menu">
              {languages.map((language, index: number) => (
                <button
                  key={index}
                  className="dropdown-item"
                  onClick={() => i18next.changeLanguage(language.code)}
                  disabled={currentLanguageCode === language.code}
                  data-testid={`changeLanguageBtn${index}`}
                >
                  <span
                    className={`flag-icon flag-icon-${language.country_code} mr-2`}
                  ></span>
                  {language.name}
                </button>
              ))}
            </div>
          </div>
          <button
            className={styles.logoutbtn}
            data-testid="logoutBtn"
            onClick={logout}
          >
            {t('logout')}
          </button>
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};

export default ListNavbar;
