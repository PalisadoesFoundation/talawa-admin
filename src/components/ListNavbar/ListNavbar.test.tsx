import React from 'react';
import { act, render, screen } from '@testing-library/react';
import { I18nextProvider } from 'react-i18next';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import 'jest-localstorage-mock';

import ListNavbar from './ListNavbar';
import i18nForTest from 'utils/i18nForTest';

async function wait(ms = 100) {
  await act(() => {
    return new Promise((resolve) => {
      setTimeout(resolve, ms);
    });
  });
}

describe('Testing List navbar component', () => {
  test('Component should be rendered properly', () => {
    localStorage.setItem('UserType', 'SUPERADMIN');
    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18nForTest}>
          <ListNavbar />
        </I18nextProvider>
      </BrowserRouter>
    );

    expect(screen.getByText('Talawa Admin Portal')).toBeInTheDocument();
    expect(screen.getByText('Roles')).toBeInTheDocument();
    expect(screen.getByText('Requests')).toBeInTheDocument();
  });

  test('Testing change language functionality', async () => {
    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18nForTest}>
          <ListNavbar />
        </I18nextProvider>
      </BrowserRouter>
    );

    await wait();

    userEvent.click(screen.getByTestId('changeLanguageBtn1'));
  });

  test('Testing when language cookie is not set', async () => {
    Object.defineProperty(window.document, 'cookie', {
      writable: true,
      value: 'i18next=',
    });

    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18nForTest}>
          <ListNavbar />
        </I18nextProvider>
      </BrowserRouter>
    );

    await wait();
  });

  test('Testing logout functionality', async () => {
    window.location.assign('/');

    render(
      <BrowserRouter>
        <I18nextProvider i18n={i18nForTest}>
          <ListNavbar />
        </I18nextProvider>
      </BrowserRouter>
    );

    userEvent.click(screen.getByTestId('logoutBtn'));
  });
});
