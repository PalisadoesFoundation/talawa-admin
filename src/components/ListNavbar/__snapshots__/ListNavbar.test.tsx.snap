// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing List navbar component testing using snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbarbgOrglist navbar navbar-expand-xl navbar-light"
      >
        <span
          class="navbarBrand navbar-brand"
        >
          <a
            class="logo"
            href="/orglist"
          >
            <img
              src="talawa-logo-200x200.png"
            />
            <strong>
              Talawa Portal
            </strong>
          </a>
        </span>
        <button
          aria-controls="navbarScroll"
          aria-label="Toggle navigation"
          class="navbar-toggler collapsed"
          type="button"
        >
          <span
            class="navbar-toggler-icon"
          />
        </button>
        <div
          class="navbar-collapse collapse"
          id="navbarScroll"
        >
          <div
            class="me-auto navbar-nav"
          />
          <div
            class="ml-auto navbar-nav"
          >
            <div
              class="dropdown mr-4"
            >
              <button
                aria-expanded="false"
                class="btn dropdown-toggle languageBtn"
                data-toggle="dropdown"
                title="Change Langauge"
                type="button"
              >
                <i
                  class="fas fa-globe"
                />
              </button>
              <div
                class="dropdown-menu"
              >
                <button
                  class="dropdown-item"
                  data-testid="changeLanguageBtn0"
                  disabled=""
                >
                  <span
                    class="flag-icon flag-icon-gb mr-2"
                  />
                  English
                </button>
                <button
                  class="dropdown-item"
                  data-testid="changeLanguageBtn1"
                >
                  <span
                    class="flag-icon flag-icon-fr mr-2"
                  />
                  Français
                </button>
                <button
                  class="dropdown-item"
                  data-testid="changeLanguageBtn2"
                >
                  <span
                    class="flag-icon flag-icon-in mr-2"
                  />
                  हिन्दी
                </button>
                <button
                  class="dropdown-item"
                  data-testid="changeLanguageBtn3"
                >
                  <span
                    class="flag-icon flag-icon-ar mr-2"
                  />
                  Español
                </button>
                <button
                  class="dropdown-item"
                  data-testid="changeLanguageBtn4"
                >
                  <span
                    class="flag-icon flag-icon-cn mr-2"
                  />
                  中國人
                </button>
              </div>
            </div>
            <button
              class="btn btn-success logoutbtn"
              data-testid="logoutBtn"
            >
              Logout
            </button>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbarbgOrglist navbar navbar-expand-xl navbar-light"
    >
      <span
        class="navbarBrand navbar-brand"
      >
        <a
          class="logo"
          href="/orglist"
        >
          <img
            src="talawa-logo-200x200.png"
          />
          <strong>
            Talawa Portal
          </strong>
        </a>
      </span>
      <button
        aria-controls="navbarScroll"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="navbarScroll"
      >
        <div
          class="me-auto navbar-nav"
        />
        <div
          class="ml-auto navbar-nav"
        >
          <div
            class="dropdown mr-4"
          >
            <button
              aria-expanded="false"
              class="btn dropdown-toggle languageBtn"
              data-toggle="dropdown"
              title="Change Langauge"
              type="button"
            >
              <i
                class="fas fa-globe"
              />
            </button>
            <div
              class="dropdown-menu"
            >
              <button
                class="dropdown-item"
                data-testid="changeLanguageBtn0"
                disabled=""
              >
                <span
                  class="flag-icon flag-icon-gb mr-2"
                />
                English
              </button>
              <button
                class="dropdown-item"
                data-testid="changeLanguageBtn1"
              >
                <span
                  class="flag-icon flag-icon-fr mr-2"
                />
                Français
              </button>
              <button
                class="dropdown-item"
                data-testid="changeLanguageBtn2"
              >
                <span
                  class="flag-icon flag-icon-in mr-2"
                />
                हिन्दी
              </button>
              <button
                class="dropdown-item"
                data-testid="changeLanguageBtn3"
              >
                <span
                  class="flag-icon flag-icon-ar mr-2"
                />
                Español
              </button>
              <button
                class="dropdown-item"
                data-testid="changeLanguageBtn4"
              >
                <span
                  class="flag-icon flag-icon-cn mr-2"
                />
                中國人
              </button>
            </div>
          </div>
          <button
            class="btn btn-success logoutbtn"
            data-testid="logoutBtn"
          >
            Logout
          </button>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
